spec/predicates/scai.md

```diff
@@ -0,0 +1,334 @@
+
+Version: 0.1
+
+Author: Marcela Melara <marcela.melara@intel.com>
+
+## Purpose
+
+The Software Supply Chain Attribute Integrity, or SCAI (pronounced "sky"), specification proposes a data
+format for capturing functional attribute and integrity information about software artifacts and their supply
+chain. SCAI data can be associated with executable binaries, statically- or dynamically-linked libraries,
+software packages, container images, software toolchains, and compute environments.
+
+Existing supply chain data formats do not capture any information about the security functionality or
+behavior of the resulting software artifact, nor do they provide sufficient evidence to support any claims of
+integrity of the supply chain processes they describe. The SCAI data format is designed to bridge this gap.
+
+For more details, see the [SCAI v0.1 specification document](https://arxiv.org/pdf/2210.05813.pdf)
+
+## Prerequisites
+
+In addition to the in-toto attestation framework,
+SCAI assumes that implementers have appropriate processes and tooling in place for
+capturing other types of software supply chain metadata, which can be extended to add support for SCAI.
+
+## Model
+
+SCAI metadata, referred to as an Attribute Assertion, describes functional attributes of a software
+artifact and its supply chain, capable of covering the full software stack of the
+toolchain that produced the artifact down to the hardware platform. SCAI Attribute Assertions include
+information about the conditions under which certain functional attributes arise, as well as (authenticated)
+evidence for the asserted attributes. The set of Assertions about a subject artifact and its producer
+is referred to as the **Attribute Report**.
+
+SCAI is intended to be implemented as part of an existing software supply chain attestation
+framework by software development tools or services (e.g., builders, CI/CD pipelines, software analysis tools)
+seeking to capture more granular information about the attributes and behavior of the software artifacts they
+produce. As such, we envision SCAI metadata being explicitly bound to, or embedded within, other metadata
+objects.
+
+## Schema
+
+The core metadata in SCAI is the Attribute Assertion. A collection of Attribute Assertions
+for a specific supply chain step or operation are issued together in a SCAI Attribute Report predicate.
+
+```
+{
+    "predicateType": "scai/attribute-report/v0.1",
+    "predicate": {
+        "subjectAttributes": [{
+            "attribute": "<ATTRIBUTE>",
+            "target": { // optional
+                "name": "<NAME>",
+                "digest": { "<ALGORITHM>": "<HEX VALUE>", ... },
+                "locationURI": "<URI>",
+                "objectType": "<OBJECT TYPE>" // optional
+            },
+            "conditions": { /* object */ }, // optional
+            "evidence": { /* object */ } // optional
+        }],
+        "producerAttributes": [{
+            "attribute": "<ATTRIBUTE>",
```

Comment:
The attribute field for the producer should be optional as it might be either the subject or producer or both.

Reply:
I'm not sure I understand what you mean. I had split the fields in the predicate by subject and producer already, but is there another scenario you have in mind?

Reply:
Disregard. I was thinking about it wrong.